gwt-localizable-processor/src/main/java/org/gwtproject/i18n/processor/LocalizableProcessingStep.java

```diff
@@ -47,105 +55,378 @@ public LocalizableProcessingStep(ProcessingEnvironment processingEnv) {
+                                    .build());
+                          });
+
+                    } else if (context.isArray(constantMethod.getMethod().getReturnType())
+                        && context.isString(
+                            context.arrayComponentType(
+                                constantMethod.getMethod().getReturnType()))) {
+                      addCacheField(constantImplBuilder, cacheAdded);
+                      Optional<CodeBlock> methodBody =
+                          getArrayMethodBody(constantMethod, locale, localeResourceMap);
+                      methodBody.ifPresent(
+                          codeBlock -> {
+                            constantImplBuilder.addMethod(
+                                MethodSpec.methodBuilder(
+                                        constantMethod.getMethod().getSimpleName().toString())
+                                    .addModifiers(Modifier.PUBLIC)
+                                    .addAnnotation(Override.class)
+                                    .returns(
+                                        TypeName.get(constantMethod.getMethod().getReturnType()))
+                                    .addCode(codeBlock)
+                                    .build());
+                          });
+
+                    } else {
+                      constantImplBuilder.addMethod(
+                          MethodSpec.methodBuilder(
+                                  constantMethod.getMethod().getSimpleName().toString())
+                              .addModifiers(Modifier.PUBLIC)
+                              .addAnnotation(Override.class)
+                              .returns(TypeName.get(constantMethod.getMethod().getReturnType()))
+                              .addCode(
+                                  getValueExpression(
+                                      context, constantMethod, locale, localeResourceMap))
+                              .build());
+                    }
+                  });
+
+          if (context.isAssignableFrom(element, ConstantsWithLookup.class)) {
+            lookupMethodsBuilder
+                .lookupMethods()
+                .forEach(methodBuilder -> constantImplBuilder.addMethod(methodBuilder.build()));
+          }
+
+          writeSource(constantImplBuilder, context, element);
+        });
+  }
+
+  private String getConstantClassName(Element element, String locale) {
+    String enclosingElementName =
+        !(ElementKind.PACKAGE == element.getEnclosingElement().getKind())
+            ? element.getEnclosingElement().getSimpleName().toString() + "_"
+            : "";
+    return enclosingElementName + element.getSimpleName().toString() + "_" + localePostfix(locale);
+  }
+
+  private String localePostfix(String locale) {
+    return "default".equals(locale) ? "" : locale;
+  }
+
+  private void addCacheField(TypeSpec.Builder constantImplBuilder, boolean[] cacheAdded) {
+    if (!cacheAdded[0]) {
```

Comment:
might be nicer to take a bool param and return a boolean value... or just don't call this method at all if cacheAdded[0] was true...?